commit b1891f7b0018e7e778f49c1817692e1f85a1f529
Author: Eric Anholt <eric@anholt.net>
Date:   Wed Nov 19 12:06:38 2014 -0800

    drm: Put an optional field in the driver struct for GEM obj struct size.
    
    This allows a driver to derive from the CMA object without copying all
    of the code.
    
    Signed-off-by: Eric Anholt <eric@anholt.net>

diff --git a/drivers/gpu/drm/drm_gem_cma_helper.c b/drivers/gpu/drm/drm_gem_cma_helper.c
index e109b49..1c25500 100644
--- a/drivers/gpu/drm/drm_gem_cma_helper.c
+++ b/drivers/gpu/drm/drm_gem_cma_helper.c
@@ -58,8 +58,11 @@ __drm_gem_cma_create(struct drm_device *drm, size_t size)
 	struct drm_gem_cma_object *cma_obj;
 	struct drm_gem_object *gem_obj;
 	int ret;
+	size_t obj_size = (drm->driver->gem_obj_size ?
+			   drm->driver->gem_obj_size :
+			   sizeof(*cma_obj));
 
-	cma_obj = kzalloc(sizeof(*cma_obj), GFP_KERNEL);
+	cma_obj = kzalloc(obj_size, GFP_KERNEL);
 	if (!cma_obj)
 		return ERR_PTR(-ENOMEM);
 
diff --git a/include/drm/drmP.h b/include/drm/drmP.h
index 0a271ca..54f5469 100644
--- a/include/drm/drmP.h
+++ b/include/drm/drmP.h
@@ -639,6 +639,7 @@ struct drm_driver {
 
 	u32 driver_features;
 	int dev_priv_size;
+	size_t gem_obj_size;
 	const struct drm_ioctl_desc *ioctls;
 	int num_ioctls;
 	const struct file_operations *fops;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1111_b1891f7b0018e7e778f49c1817692e1f85a1f529.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1111_b1891f7b0018e7e778f49c1817692e1f85a1f529.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
